telosnetwork/telos-wallet
# resolve real gas estimation

# Fixes #274 

## Description
Currently, we are using a couple of constants to pass the not-so-estimated gas limit to the function getEstimatedGas which converts that limit to the actual token and fiat costs.
This PR includes a dynamic way of estimating the cost of gas by creating a transaction simulation and asking the contract for a gas estimation limit. Also, a loading flag was used to show the progress of that process.

https://github.com/telosnetwork/telos-wallet/blob/223-send-page-api-integration-incorporate-support-for-sending-evm-tokens/src/pages/evm/wallet/SendPage.vue#L199-L201

Here is the new code
https://github.com/telosnetwork/telos-wallet/blob/274-resolve-real-gas-estimation/src/pages/evm/wallet/SendPage.vue#L256-L258

Which finally uses this function to get the Gas limit estimation
https://github.com/telosnetwork/telos-wallet/blob/274-resolve-real-gas-estimation/src/antelope/chains/EVMChainSettings.ts#L249-L265


## Test scenarios
- go to [this link](https://deploy-preview-282--wallet-staging.netlify.app/)
- login on EVM
- press Send button
  - you should see a spinner next to the estimated gas title while it it processing
  - the spinner should stop and show to real gas estimation
  - everytime you change the selected token it will trigger the gas estimation again



## Checklist:
